net-irc/weechat/weechat-4.0.4.ebuild

```diff
@@ -0,0 +1,198 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+LUA_COMPAT=( lua5-{1..4} )
+PYTHON_COMPAT=( python3_{9..11} )
+
+inherit cmake lua-single python-single-r1 xdg-utils
+
+if [[ ${PV} == "9999" ]] ; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/weechat/weechat.git"
+else
+	inherit verify-sig
+	SRC_URI="https://weechat.org/files/src/${P}.tar.xz
+		verify-sig? ( https://weechat.org/files/src/${P}.tar.xz.asc )"
+	VERIFY_SIG_OPENPGP_KEY_PATH=${BROOT}/usr/share/openpgp-keys/weechat.org.asc
+	BDEPEND+="verify-sig? ( sec-keys/openpgp-keys-weechat )"
+	KEYWORDS="~amd64 ~arm ~arm64 ~ppc ~ppc64 ~riscv ~x86 ~x64-macos"
+fi
+
+DESCRIPTION="Portable and multi-interface IRC client"
+HOMEPAGE="https://weechat.org/"
+
+LICENSE="GPL-3"
+SLOT="0/${PV}"
+
+NETWORKS="+irc"
+PLUGINS="+alias +buflist +charset +exec +fifo +fset +logger +relay +scripts +spell +trigger +typing +xfer"
+# dev-lang/v8 was dropped from Gentoo so we can't enable javascript support
+# dev-lang/php eclass support is lacking, php plugins don't work. bug #705702
+SCRIPT_LANGS="guile lua +perl +python ruby tcl"
+LANGS=" cs de es fr it ja pl ru"
+IUSE="doc enchant man nls selinux test ${SCRIPT_LANGS} ${PLUGINS} ${INTERFACES} ${NETWORKS}"
+
+REQUIRED_USE="
+	enchant? ( spell )
+	lua? ( ${LUA_REQUIRED_USE} )
+	python? ( ${PYTHON_REQUIRED_USE} )
+	test? ( nls )
+"
+
+RDEPEND="
+	app-arch/zstd:=
+	dev-libs/libgcrypt:0=
+	net-libs/gnutls:=
+	sys-libs/ncurses:0=
+	sys-libs/zlib:=
+	net-misc/curl[ssl]
+	charset? ( virtual/libiconv )
+	guile? ( >=dev-scheme/guile-2.0:12= )
+	lua? ( ${LUA_DEPS} )
+	nls? ( virtual/libintl )
+	perl? (
+		dev-lang/perl:=
+		virtual/libcrypt:=
```

Comment:
There is no need to explicitly add `virtual/libcrypt:=` as a dependency, since `dev-lang/perl` already depends on it unconditionally, which means it will be pulled in whenever Perl is installed.

Reply:
No, that's not right. You should explicitly depend on anything you need, _especially_ if there's a subslot dep involved (but not just then).

Reply:
My bad. I've checked and @thesamesam is actually right (and thus @sevz17 too):

```
$ ldd /usr/lib64/weechat/plugins/perl.so
        linux-vdso.so.1 (0x00007ffe6fb61000)
        libperl.so.5.38 => /usr/lib64/libperl.so.5.38 (0x00007f3326ceb000)
        libc.so.6 => /lib64/libc.so.6 (0x00007f3326b0c000)
        libm.so.6 => /lib64/libm.so.6 (0x00007f3326a2f000)
        libcrypt.so.2 => /lib64/libcrypt.so.2 (0x00007f33269f4000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f33270ea000)
```

This comment can be ignored.